logstash-core/src/test/java/org/logstash/jackson/StreamReadConstraintsUtilTest.java

```diff
@@ -0,0 +1,205 @@
+                    .returns(defaults.getMaxStringLength(), from(StreamReadConstraints::getMaxStringLength))
+                    .returns(defaults.getMaxNumberLength(), from(StreamReadConstraints::getMaxNumberLength));
+
+            assertThatThrownBy(configuredUtil::validateIsGlobalDefault).isInstanceOf(IllegalStateException.class).hasMessageContaining(MAX_NESTING_DEPTH.getPropertyName());
+
+            configuredUtil.applyAsGlobalDefault();
+            assertThatCode(configuredUtil::validateIsGlobalDefault).doesNotThrowAnyException();
+        });
+    }
+
+    @Test
+    public void configuresMaxNestingDepthInvalid() {
+        final Properties properties = new Properties();
+        properties.setProperty(MAX_NESTING_DEPTH.getPropertyName(), "NaN");
+
+        fromProperties(properties, (configuredUtil, defaults) -> {
+            assertThatThrownBy(configuredUtil::get)
+                    .isInstanceOf(IllegalArgumentException.class)
+                    .hasMessageContaining(MAX_NESTING_DEPTH.getPropertyName());
+        });
+    }
+
+    @Test
+    public void configuresMaxNestingDepthNegative() {
+        final Properties properties = new Properties();
+        properties.setProperty(MAX_NESTING_DEPTH.getPropertyName(), "-1000");
+
+        fromProperties(properties, (configuredUtil, defaults) -> {
+            assertThatThrownBy(configuredUtil::get)
+                    .isInstanceOf(IllegalArgumentException.class)
+                    .hasMessageContaining(MAX_NESTING_DEPTH.getPropertyName());
+        });
+    }
+
+    @Test
+    public void validatesApplication() {
+        final Properties properties = new Properties();
+        properties.setProperty(StreamReadConstraintsUtil.PROP_PREFIX + "unsupported-option1", "100");
+        properties.setProperty(StreamReadConstraintsUtil.PROP_PREFIX + "unsupported-option2", "100");
+        properties.setProperty(MAX_NESTING_DEPTH.getPropertyName(), "1010");
+        properties.setProperty(MAX_STRING_LENGTH.getPropertyName(), "1011");
+        properties.setProperty(MAX_NUMBER_LENGTH.getPropertyName(), "1110");
+
+        fromProperties(properties, (configuredUtil, defaults) -> {
+            configuredUtil.applyAsGlobalDefault();
+            configuredUtil.validateIsGlobalDefault();
+        });
+
+        assertLogObserved(Level.INFO, "override `" + MAX_NESTING_DEPTH.getPropertyName() + "` configured to `1010`");
+        assertLogObserved(Level.INFO, "override `" + MAX_STRING_LENGTH.getPropertyName() + "` configured to `1011`");
+        assertLogObserved(Level.INFO, "override `" + MAX_NUMBER_LENGTH.getPropertyName() + "` configured to `1110`");
+
+        assertLogObserved(Level.WARN, "override `" + StreamReadConstraintsUtil.PROP_PREFIX + "unsupported-option1` is unknown and has been ignored");
+        assertLogObserved(Level.WARN, "override `" + StreamReadConstraintsUtil.PROP_PREFIX + "unsupported-option1` is unknown and has been ignored");
```

Comment:
Mabye copy paste error? Should this be looking for `option2`?
```suggestion
        assertLogObserved(Level.WARN, "override `" + StreamReadConstraintsUtil.PROP_PREFIX + "unsupported-option2` is unknown and has been ignored");
```